pkg/maps/configmap/config_map.go

```diff
@@ -0,0 +1,181 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package configmap
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+	"unsafe"
+
+	"github.com/cilium/cilium/pkg/bpf"
+	"github.com/cilium/cilium/pkg/controller"
+)
+
+const (
+	// MapName name of map used to pin map for datapath
+	MapName = "cilium_runtime_config"
+
+	// MaxEntries represents the maximum number of config entries.
+	// Initially defined as 256, so that downgrade from a future version having more than one
+	// entry works without necessarily resizing the map. Entries not known by the datapath
+	// version are simply ignored.
+	// Currently only one is used. Each new entry needs it's own struct/layout
+	// of the same size (64 bytes each).
+	MaxEntries = 256
+)
+
+// ConfigKey is the index to the Unix time offset. Only value 0 is supported for now.
+// +k8s:deepcopy-gen=true
+// +k8s:deepcopy-gen:interfaces=github.com/cilium/cilium/pkg/bpf.MapKey
+type ConfigKey struct {
```

Comment:
Why are the `Config`* types exported? Seems like nothing is using it externally.